xdsl/ir.py

```diff
@@ -524,8 +534,29 @@ def irdl_definition(cls) -> OpDef:
         ...
 
 
-@dataclass(eq=False)
-class Block:
+@dataclass
+class BinaryOperation(Operation):
```

Comment:
I think you might be able to give this class the fields, just like we do for other operations, just do not add the decorator. Then, you can get rid of the asserts in the verifier.

Reply:
I'm not sure you want to do that, since then the fields will not have the proper types (since you might sometimes want to have an `IntegerType`, but sometimes else a `Vector`.

Reply:
Yeah, agreed :thinking: 
Not sure how to circumvent that. Maybe define them initially without a type constraint and them override them in the child with the type constraint?

Reply:
what fields do you mean?

Reply:
He means the fields you have to add, if you were to follow my proposal.